packages/date-time-utilities/src/dateGrid/getBoundedDateRange.ts

```diff
@@ -8,11 +8,12 @@ import { compareDatePart } from '../dateMath/dateMath';
  */
 export const getBoundedDateRange = (dateRange: Date[], minDate?: Date, maxDate?: Date): Date[] => {
   let boundedDateRange = [...dateRange];
+  const zeroVal: Number = Number(0);
   if (minDate) {
-    boundedDateRange = boundedDateRange.filter((date: Date) => compareDatePart(date, minDate as Date) >= 0);
+    boundedDateRange = boundedDateRange.filter((date: Date) => compareDatePart(date, minDate as Date) >= zeroVal);
   }
   if (maxDate) {
-    boundedDateRange = boundedDateRange.filter((date: Date) => compareDatePart(date, maxDate as Date) <= 0);
+    boundedDateRange = boundedDateRange.filter((date: Date) => compareDatePart(date, maxDate as Date) <= zeroVal);
```

Comment:
Fixes like this probably should go in the master branch instead of xplat. That way we don't have to worry about merge conflicts in unrelated code when updating the branch.

Also, I think the fix here is to change the return type of `compareDatePart` to lower-case `number`, since it's returning the difference of two lower-case `number` values. I'm guessing it was a typo to use upper-case `Number`.

Reply:
I made this switch. For other typescript 5 changes I agree we can do them in master.